### DISCUSSION TITLE
Workers development in Git Codespaces

### DISCUSSION THREAD

**Discussion by maildebjyoti** [cloudflare/workers-sdk#2763] [General]
I was trying to evaluate the Workers in Codespaces but found that it fails due to the Oauth2.0 callback. Tried searching for a workaround by passing the API keys but the documentation is not clear to me and seems like it is pointing to v1.

The approach I took was to run "_**wrangler dev --local**_" and then made the **_port public_**. But I am also interested to deploy the same. Following this --> _**https://developers.cloudflare.com/workers/wrangler/ci-cd/**_ the **_"wrangler publish"_** works but the **_"wrangler login"_** still fails.

Did anyone try something similar or any blogs which I can refer to?

**Comment by penalosa** [comment]
`wrangler login` won't work in GitHub codespaces due to the Oauth requirement. Using a generated API token as outlined in https://developers.cloudflare.com/workers/wrangler/ci-cd/ is the right approach. To run wrangler commands, prefix the command with `CLOUDFLARE_API_TOKEN`. For example (replace `<TOKEN>` with your actual API token):

```sh
CLOUDFLARE_API_TOKEN=<TOKEN> npx wrangler dev
```

**Reply by maildebjyoti** [reply]
Thanks!

**Comment by souphil** [comment]
npx wrangler login --device --browser=false
